encode.go

```diff
@@ -30,6 +31,26 @@ func newEncoder() (e *encoder) {
 	return e
 }
 
+func newFileEncoder(w io.Writer) (e *encoder) {
+	e = &encoder{}
+	e.must(yaml_emitter_initialize(&e.emitter))
+	yaml_emitter_set_output_file(&e.emitter, w)
+	yaml_emitter_set_unicode(&e.emitter, true)
+	e.must(yaml_stream_start_event_initialize(&e.event, yaml_UTF8_ENCODING))
```

Comment:
A similar issue applies here to the one in newFileParser. I don't think this should actually write anything until a value is encoded.